pages/api/preview.js

```diff
@@ -47,14 +48,31 @@ const preview = async (request, response) => {
     return response.status(401).send({ message: "Invalid uri" });
   }
 
-  const redirectUri = res.entry.uri === CRAFT_HOMEPAGE_URI ? "" : res.entry.uri;
-  const redirect = `/${locale}/${redirectUri}`;
-  const cookiePath = `${site === "default" ? "" : `/${locale}`}/${redirectUri}`;
-
   // Enable Draft Mode by setting the cookie
   response.setDraftMode({ enable: true });
-  response.setPreviewData({ previewToken }, { path: cookiePath, maxAge: 120 });
-  response.redirect(redirect);
+
+  if (res.entry.uri === CRAFT_HOMEPAGE_URI) {
```

Comment:
refine the preview URI structure to reflect that the homepage is now in it's own directory